### ISSUE TITLE
Best practices jit and vmap

### ISSUE THREAD

**Issue by GJBoth** [google/jax#5199]
Hi jax team,

I'm working on switching our code to Jax (using Flax as NN library) and I'm amazed with jit and vmap. I'm wondering if there are best practices for *when* to apply these. For example:
1. I create a loss function and then use jax.value_and_grad(loss) Should I jit the loss, the value_and_grad, both ,or doesn't it matter? 
2. If I use vmap to calculate the loss, should I first calculate the forward pass and then vmap for the loss, or can I vmap the forward pass well? Is there a performance penalty for this?

Maybe these things don't matter at all and the jit and vmap magic is super robust against amateurs like me, but otherwise a simple best practices page could help. Thanks!

**Comment by 8bitmp3** [comment]
@GJBoth I'd also love to know the best practices - good questions. Maybe this issue should be moved [GitHub Discussions](https://github.com/google/jax/discussions) by the JAX admins (there's a guide for this: https://docs.github.com/en/free-pro-team@latest/discussions/managing-discussions-for-your-community/moderating-discussions#converting-an-issue-to-a-discussion) cc Flax @avital
